libcloud/compute/drivers/vsphere.py

```diff
@@ -49,506 +36,228 @@
+                        raise InvalidCredsError('Check your username and '
+                                                'password are valid')
+                    if 'connection refused' in error_message or \
+                            'is not a vim server' in error_message:
+                        raise Exception('Check that the host provided '
+                                        'is a vSphere installation')
+                    if 'name or service not known' in error_message:
+                        raise Exception('Check that the vSphere host is '
+                                        'accessible')
+                    raise Exception('Cannot connect to vSphere using '
+                                    'self signed certs')
             else:
-                raise NotImplementedError('Unsupported API version: %s' %
-                                          (api_version))
-        return super(VSphereNodeDriver, cls).__new__(cls)
+                raise Exception('Cannot connect to vSphere')
 
-    def __init__(self, username, password, secure=True,
-                 host=None, port=None, url=None, timeout=None):
-        self.url = url
-        super(VSphereNodeDriver, self).__init__(key=username, secret=password,
-                                                secure=secure, host=host,
-                                                port=port, url=url)
-
-    @wrap_non_libcloud_exceptions
     def list_locations(self):
         """
-        List available locations.
-
-        In vSphere case, a location represents a datacenter.
+        Lists locations
         """
-        datacenters = self.connection.client.get_datacenters()
-
-        locations = []
-        for id, name in datacenters.items():
-            location = NodeLocation(id=id, name=name, country=None,
-                                    driver=self)
-            locations.append(location)
+        return []
 
-        return locations
+    def list_sizes(self):
+        """
+        Lists sizes
+        """
+        return []
 
-    @wrap_non_libcloud_exceptions
     def list_images(self):
         """
-        List available images (templates).
+        Lists images
         """
-        server = self.connection.client
-
-        names = ['name', 'config.uuid', 'config.template']
-        properties = server._retrieve_properties_traversal(
-            property_names=names,
-            from_node=None,
-            obj_type=MORTypes.VirtualMachine)
-
-        images = []
-        for prop in properties:
-            id = None
-            name = None
-            is_template = False
-
-            for item in prop.PropSet:
-                if item.Name == 'config.uuid':
-                    id = item.Val
-                if item.Name == 'name':
-                    name = item.Val
-                elif item.Name == 'config.template':
-                    is_template = item.Val
+        return []
```

Comment:
instead of returning nothing, these should fall back to the base class so that in provider tables it shows as not supported